pymongo/common.py

```diff
@@ -584,10 +607,61 @@ def validate_tzinfo(dummy, value):
     'server_selector': validate_is_callable_or_none,
 }
 
-URI_VALIDATORS.update(TIMEOUT_VALIDATORS)
-VALIDATORS = URI_VALIDATORS.copy()
+# Dictionary where keys are any URI option name, and values are the
+# internally-used names of that URI option. Options with only one name
+# variant need not be included here. Options whose public and internal
+# names are the same need not be included here.
+INTERNAL_URI_OPTION_NAME_MAP = {
+    'j': 'journal',
+    'wtimeout': 'wtimeoutms',
+    'tls': 'ssl',
+    'tlsallowinvalidcertificates': 'ssl_cert_reqs',
+    'tlsallowinvalidhostnames': 'ssl_match_hostname',
+    'tlscrlfile': 'ssl_crlfile',
+    'tlscafile': 'ssl_ca_certs',
+    'tlsclientcertfile': 'ssl_certfile',
+    'tlsclientkeypassword': 'ssl_pem_passphrase',
+    'tlsclientkeyfile': 'ssl_keyfile',
+}
+
+# Map from deprecated URI option names to the updated option names.
+# Case is preserved for updated option names as they are part of user warnings.
+URI_OPTIONS_DEPRECATION_MAP = {
+    'j': 'journal',
+    'wtimeout': 'wTimeoutMS',
+    'ssl': 'tls',
+    'ssl_cert_reqs': 'tlsAllowInvalidCertificates',
+    'ssl_match_hostname': 'tlsAllowInvalidHostnames',
+    'ssl_crlfile': 'tlsCRLFile',
+    'ssl_ca_certs': 'tlsCAFile',
+    'ssl_certfile': 'tlsClientCertFile',
```

Comment:
Because of the naming conflict we had with the server, we're only providing a URI option to support a unified file that provides both the cert chain and the private key. Let's not deprecate these three options. There are definitely users that want to store the private key in a separate file.

Reply:
Did you mean to say two options? AFAICT we only need to not deprecate the `ssl_certfile` and `ssl_keyfile` options. Everything else remains the same, is it not?

Reply:
Yes, you're right.